### PR TITLE
[DOCS] Adds API docs for bert_ja text embedding tokenizer option

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

Related PR: https://github.com/elastic/elasticsearch/pull/95546
This PR adds documentation of the `bert_ja` tokenizer option to all task types that has `bert` as a tokenizer option.
It also marks `xlm_roberta` as experimental in the list of tokenizers.

### Preview

* [PUT trained models](https://elasticsearch_96873.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-trained-models.html)
* [GET trained models](https://elasticsearch_96873.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-trained-models.html)
* [INFER trained model](https://elasticsearch_96873.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/infer-trained-model.html) 

Screenshot 2023-06-15 at 16 19 50

Screenshot 2023-06-15 at 16 20 18

### CODE REVIEW

**Review by dimkots — APPROVED** [review]

**Comment by daixque** [comment]
- `bert-ja` is wrong. Please use `bert_ja` instead.
- As same as `bert`, `bert_ja` is not only for text embedding, but also the other tasks as well.

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -944,14 +944,34 @@ values are
...
+Indicates the tokenization to perform and the desired settings.
+The default tokenization configuration is `bert`. Valid tokenization
+values are
++
+--
+* `bert`: Use for BERT-style models
+* `mpnet`: Use for MPNet-style models
+* `roberta`: Use for RoBERTa-style and BART-style models
+* experimental:[] `xlm_roberta`: Use for XLMRoBERTa-style models
+* experimental:[] `bert-ja`: Use for BERT-style models trained for the Japanese
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc:961]:
```suggestion
* experimental:[] `bert_ja`: Use for BERT-style models trained for the Japanese
```

File: docs/reference/ml/trained-models/apis/get-trained-models.asciidoc [hunk 2]
```diff
@@ -787,6 +787,33 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenizati
 (Optional, boolean)
 include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-roberta-with-special-tokens]
 ========
+`bert-ja`::::
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/get-trained-models.asciidoc:790]:
```suggestion
`bert_ja`::::
```

File: docs/reference/ml/trained-models/apis/get-trained-models.asciidoc [hunk 3]
```diff
@@ -787,6 +787,33 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenizati
...
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-ja]
++
+.Properties of bert-ja
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/get-trained-models.asciidoc:794]:
```suggestion
.Properties of bert_ja
```

File: docs/reference/ml/trained-models/apis/infer-trained-model.asciidoc [hunk 4]
```diff
@@ -564,6 +564,17 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenizati
 (Optional, string)
 include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-truncate]
 =======
+`bert-ja`::::
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/infer-trained-model.asciidoc:567]:
```suggestion
`bert_ja`::::
```

File: docs/reference/ml/trained-models/apis/infer-trained-model.asciidoc [hunk 5]
```diff
@@ -564,6 +564,17 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenizati
...
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-ja]
++
+.Properties of bert-ja
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/infer-trained-model.asciidoc:571]:
```suggestion
.Properties of bert_ja
```

File: docs/reference/ml/trained-models/apis/put-trained-models.asciidoc [hunk 6]
```diff
@@ -1145,6 +1145,33 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenizati
 (Optional, boolean)
 include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-roberta-with-special-tokens]
 =======
+`bert-ja`::::
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/put-trained-models.asciidoc:1148]:
```suggestion
`bert_ja`::::
```

File: docs/reference/ml/trained-models/apis/put-trained-models.asciidoc [hunk 7]
```diff
@@ -1145,6 +1145,33 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenizati
...
+(Optional, object)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization-bert-ja]
++
+.Properties of bert-ja
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/put-trained-models.asciidoc:1152]:
```suggestion
.Properties of bert_ja
```

**Comment by szabosteve** [comment]
@daixque Thanks for the feedback. I updated the PR accordingly.

**Comment by daixque** [comment]
LGTM

**Pull request merged**